engine/src/flutter/impeller/renderer/backend/vulkan/descriptor_pool_vk.cc

```diff
@@ -130,30 +109,35 @@ fml::Status DescriptorPoolVK::CreateNewPool(const ContextVK& context_vk) {
   return fml::Status();
 }
 
-void DescriptorPoolRecyclerVK::Reclaim(vk::UniqueDescriptorPool&& pool) {
+void DescriptorPoolRecyclerVK::Reclaim(
+    DescriptorCacheMap descriptor_sets,
+    std::vector<vk::UniqueDescriptorPool> pools) {
   // Reset the pool on a background thread.
   auto strong_context = context_.lock();
   if (!strong_context) {
     return;
   }
-  auto device = strong_context->GetDevice();
-  device.resetDescriptorPool(pool.get());
 
-  // Move the pool to the recycled list.
-  Lock recycled_lock(recycled_mutex_);
+  for (auto& [_, cache] : descriptor_sets) {
+    cache.unused.insert(cache.unused.end(), cache.used.begin(),
+                        cache.used.end());
+    cache.used.clear();
+  }
 
-  if (recycled_.size() < kMaxRecycledPools) {
-    recycled_.push_back(std::move(pool));
-    return;
+  // Move the pool to the recycled list. If more than 32 pool are
+  // cached then delete the oldest entry.
+  Lock recycled_lock(recycled_mutex_);
+  if (recycled_.size() >= kMaxRecycledPools) {
+    auto& front = recycled_.front();
+    front->Destroy();
+    recycled_.erase(recycled_.begin());
```

Comment:
You are erasing from the front which is O(n), can't we remove from the end?

Reply:
Sure, we can just delete the newest entry instead.